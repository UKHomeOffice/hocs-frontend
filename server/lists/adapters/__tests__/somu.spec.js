const {
    somuTypesAdapter,
    somuItemsAdapter } = require('../somu');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Somu Adapter', () => {

    describe('Somu Type Adapter', () => {
        const mockData = [
            { uuid: '00000000-0000-0000-0000-000000000000', caseType: 'TestCaseType', type: 'TestType', schema: '{"test":1}', active: true }
        ];

        it('should transform somu type', async () => {
            const results = await somuTypesAdapter(mockData, { logger: mockLogger });

            expect(results).toBeDefined();
            expect(results).toMatchSnapshot();
        });
    });

    describe('Somu Item Adapter', () => {

        it('should transform somu type with data', async () => {
            const mockData = { uuid: '00000000-0000-0000-0000-000000000000', data: '[{"test":1}]', deleted: false };

            const results = await somuItemsAdapter(mockData, { logger: mockLogger });

            expect(results).toBeDefined();
            expect(results).toMatchSnapshot();
        });

        it('should transform somu type with deleted set as true', async () => {
            const mockData = { uuid: '00000000-0000-0000-0000-000000000000', data: null, deleted: true };

            const results = await somuItemsAdapter(mockData, { logger: mockLogger });

            expect(results).toBeDefined();
            expect(results).toMatchSnapshot();
        });
    });

});