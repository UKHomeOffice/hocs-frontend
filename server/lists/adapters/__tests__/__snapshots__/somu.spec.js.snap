// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Somu Adapter Somu Item Adapter should transform somu type with data 1`] = `
Object {
  "data": Array [
    Object {
      "test": 1,
    },
  ],
  "deleted": false,
  "uuid": "00000000-0000-0000-0000-000000000000",
}
`;

exports[`Somu Adapter Somu Item Adapter should transform somu type with deleted set as true 1`] = `
Object {
  "data": null,
  "deleted": true,
  "uuid": "00000000-0000-0000-0000-000000000000",
}
`;

exports[`Somu Adapter Somu Type Adapter should transform somu type 1`] = `
Array [
  Object {
    "key": Array [
      "TestCaseType",
      "TestType",
    ],
    "value": Object {
      "active": true,
      "caseType": "TestCaseType",
      "schema": Object {
        "test": 1,
      },
      "type": "TestType",
      "uuid": "00000000-0000-0000-0000-000000000000",
    },
  },
]
`;
